packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/e2e/pageObject.ts

```diff
@@ -0,0 +1,295 @@
+import type {Locator, Page, Request} from '@playwright/test';
+import {
+  isUaClickEvent,
+  isUaCustomEvent,
+} from '../../../../../../playwright/utils/requests';
+
+const minimumCitationTooltipDisplayDurationMs = 1500;
+
+export class GeneratedAnswerObject {
+  constructor(
+    public page: Page,
+    public streamId: string
+  ) {
+    this.page = page;
+    this.streamId = streamId;
+  }
+
+  get likeButton(): Locator {
+    return this.page.locator('[data-cy="feedback__like-button"]');
+  }
+
+  get dislikeButton(): Locator {
+    return this.page.locator('[data-cy="feedback__dislike-button"]');
+  }
+
+  get copyToClipboardButton(): Locator {
+    return this.page.locator(
+      '[data-test="generated-answer__copy-to-clipboard"]'
+    );
+  }
+
+  get toggleButton(): Locator {
+    return this.page.locator('[data-test="generated-answer__toggle-button"]');
+  }
+
+  feedbackOption(value: string, index: number): Locator {
+    return this.page
+      .locator('.feedback-modal-qna__question input', {
+        has: this.page.locator(`[value="${value}"]`),
+      })
+      .nth(index);
+  }
+
+  questionContainer(questionText: string): Locator {
+    return this.page.locator('.feedback-modal-qna__question', {
+      hasText: questionText,
+    });
+  }
+
+  answerOption(questionText: string, answerValue: string): Locator {
+    return this.questionContainer(questionText).locator('.slds-radio_button', {
+      hasText: new RegExp(`^${answerValue}$`),
+    });
+  }
+
+  get feedbackDocumentUrlInput(): Locator {
+    return this.page.locator(
+      '.feedback-modal-qna__body input[name="documentUrl"]'
+    );
+  }
+
+  get feedbackDetailsInput(): Locator {
+    return this.page.locator(
+      '.feedback-modal-qna__body [data-name="details"] textarea'
+    );
+  }
+
+  get submitFeedbackButton(): Locator {
+    return this.page.getByRole('button', {name: /Send feedback/i});
+  }
+
+  get citationLink(): Locator {
+    return this.page.locator(
+      '[data-test="generated-answer__citations"] .citation__link'
+    );
+  }
+
+  async hoverOverCitation(index: number): Promise<void> {
+    // waiting 500ms to allow the component to render completely, cause any re-rendering abort the hover action.
+    await this.page.waitForTimeout(500);
+    await this.citationLink.waitFor({state: 'visible'});
+    await this.citationLink.nth(index).hover();
```

Comment:
This already waits for "actionability" with Playwright, are you sure you need the above waitFor?

> * **Details**
   *
   * This method hovers over the element by performing the following steps:
   * 1. Wait for [actionability](https://playwright.dev/docs/actionability) checks on the element, unless
   *    [`force`](https://playwright.dev/docs/api/class-locator#locator-hover-option-force) option is set.
   * 1. Scroll the element into view if needed.